library/Fission/Types.hs

```diff
@@ -396,33 +399,48 @@ instance User.Creator Fission where
           Left err -> Error.relaxedLeft err
           Right _  -> Right userId
 
+runUpdate ::
+     Fission (Either User.Modifier.Errors a)
+  -> UserId
+  -> Text
+  -> Text
+  -> Fission (Either User.Modifier.Errors a)
+runUpdate updateDB uID subdomain txt =
+  updateDB >>= \case
+    Left err ->
+      return $ Left err
+  
+    Right dbVal -> do
+      runDB (User.getById uID) >>= \case
+        Nothing ->
+          return . Error.openLeft $ NotFound @User
+
+        Just (Entity _ User { userUsername = Username rawUN }) -> do
+          domainName <- asks userRootDomain
+          zoneID     <- asks userZoneID
+
+          let
+            url = URL {domainName, subdomain = Just (Subdomain subdomain) <> Just (Subdomain rawUN)}
+            segments = DNS.splitRecord txt
+
+          Route53.set Txt url zoneID segments 10 <&> \case
+            Left serverErr -> Error.openLeft serverErr
+            Right _        -> Right dbVal
+
+
 instance User.Modifier Fission where
   updatePassword uID pass now =
     runDB $ User.updatePassword uID pass now
 
   updatePublicKey uID pk now =
-    runDB (User.updatePublicKey uID pk now) >>= \case
-      Left err ->
-        return $ Left err
+    runUpdate (User.updatePublicKey uID pk now) uID "_did" keyTxt 
+    where 
+      keyTxt = textDisplay $ DID Key pk
```

Comment:
💅 Why bother with the `where` here? If you want a shorter line, we can break out more sections to make it more semantic. The where as a line break feels odd.

```haskell
-- Idiomatic line break:
updatePublicKey uID pk now = 
  runUpdate (User.updatePublicKey uID pk now) uID "_did" $
    textDisplay (DID Key pk)

-- Useful/clarifying wheres:
updatePublicKey uID pk now = runUpdate updatePK uID "_did" keyTxt
  where
    keyTxt = textDisplay $ DID Key pk
    updatePK = User.updatePublicKey uID pk now
```

Reply:
yeah good call. I think I actually like the second one best